### DISCUSSION TITLE
Unexpected behaviour for RuntimeConfig while having SSR and prerender pages ?

### DISCUSSION THREAD

**Discussion by duduyo** [nuxt/nuxt#27464] [Questions] [answered]
Hi,

In our app, we have  both dynamic pages and static pages.
A dynamic page is typically a search result page presenting a "product" of our catalog (let's call this page /products/{productId})
A static page is typically a help page (let's call it /help)

We also use RuntimeConfig, to inject distinct config between local, staging or prod env.
Simplified extract of nuxt.config.ts
```
runtimeConfig: {
    public: {
        apiBaseUrl: 'http://localhost:3000/api', 
    }
}
```
The nuxt config file contains the default value, suitable when running nuxt locally in dev mode
When executed in our staging or prod server, we declare the env variable NUXT_PUBLIC_API_BASE_URL=https://myprodserver:3000/api' in order to replace the default value.

My question is about the behaviour on the prod server :
- when I open the app at the product page (typing the link https://.../products/{productId} in the browser), it works, and my app use the RuntimeConfig picked in the env variable (NUXT_PUBLIC_API_BASE_URL=https://myprodserver:3000/api' )
- but when I open the app directly on the help page (typing the link https://.../help in the browser), if I log the RuntimeConfig, i see that the apiBaseUrl is equal to the default value ('http://localhost:3000/api'). If I navigate from this page to the product page (client side routing, no full reload), the product page also use the value ('http://localhost:3000/api')

My understanding is that, when using a mix of prerendered and ssr pages in the same app, the RuntimeConfig is loaded by the first page coming from the server. 
If it's a prerendered page, the RuntimeConfig will contain the default values (the only ones available at build time). And subsequent client navigation to SSR pages will keep using these values.
If it's an SSR page, the RuntimeConfig will contain default values overridden by config existing in env variables.

1) Am I right ?
2) Should I avoid to use a mix of prerender & ssr page (=> that's what we've done for the moment, using SSR everywhere - and looking at SWR and ISR options).
3) When using a mix of prerender & ssr page, the behaviour is not really consistent... Perhaps it's worth a warning in the docs ?

Thank you for your help !
And thanks for the great job on Nuxt !!! 

Lionel

**Comment by TheAlexLichter** [comment, marked as answer]
when you prerender pages, these will be "statically" baked as HTML files. there, you can't update runtimeConfig because the server is just serving the HTML.

`isr: true` would be a better option 👍
